### DISCUSSION TITLE
View operation doing a commit on DB abstraction mod

### DISCUSSION THREAD

**Discussion by eduardonunesp** [paradigmxyz/reth#5047] [Technical discussions]
I'm curious why the view operation needs to perform a commit at [db abstraction](https://github.com/paradigmxyz/reth/blame/12ac1f153ff0be850c8564f0fbce7b008e9f9adc/crates/storage/db/src/abstraction/database.rs#L37)

**Comment by mattsse** [comment]
good question 
@rakita @joshieDo

**Comment by joshieDo** [comment]
More details [here](https://libmdbx.dqdkfa.ru/usage.html#autotoc_md48). It should be aborted or committed.

We could not include it, and eventually the `drop`of the underlying db transaction would be called, but i feel like it's cleaner to be explicit about it. 

Being explicit also creates a good practice of not letting db transactions run too long, which is bad
